src/NumericalAlgorithms/Spectral/SwshCoefficients.hpp

```diff
@@ -311,5 +316,158 @@ class CoefficientsMetadata {
  * for more details on the lazy cache.
  */
 const CoefficientsMetadata& cached_coefficients_metadata(size_t l_max) noexcept;
+
+// @{
+/*!
+ * \ingroup SwshGroup
+ * \brief Compute the mode coefficient for the convention of
+ * \cite Goldberg1966uu.
+ *
+ * \details  See the documentation for `TransformJob` for complete
+ * details on the libsharp and Goldberg coefficient representations.
+ * This interface takes a `Coefficients::CoefficientsConstIterator`, so
+ * that iterating over the libsharp data structure and performing computations
+ * based on the corresponding Goldberg modes is convenient. Two functions are
+ * provided, one for the positive m modes in the Goldberg representation, and
+ * one for the negative m modes, as the distinction is not made by the
+ * coefficient iterator.
+ *
+ * \param coefficient_info An iterator which stores an \f$l\f$ and \f$|m|\f$
+ * for the mode to be converted to the Goldberg form.
+ * \param libsharp_modes The libsharp-compatible modal storage to be accessed
+ * \param radial_offset The index of which angular slice is desired. Modes for
+ * concentric spherical shells are stored as consecutive blocks of angular
+ * modes.
+ */
+template <int Spin>
+std::complex<double> libsharp_mode_to_goldberg_plus_m(
+    const LibsharpCoefficientInfo& coefficient_info,
+    const SpinWeighted<ComplexModalVector, Spin>& libsharp_modes,
+    size_t radial_offset) noexcept;
+
+template <int Spin>
+std::complex<double> libsharp_mode_to_goldberg_minus_m(
+    const LibsharpCoefficientInfo& coefficient_info,
+    const SpinWeighted<ComplexModalVector, Spin>& libsharp_modes,
+    size_t radial_offset) noexcept;
+// @}
+
+/*!
+ * \ingroup SwshGroup
+ * \brief Compute the mode coefficient for the convention of
+ * \cite Goldberg1966uu. See the documentation for `TransformJob` for complete
+ * details on the libsharp and Goldberg coefficient representations.
+ *
+ * \details This interface extracts the (`l`, `m`) Goldberg mode from the
+ * provided libsharp-compatible `libsharp_modes`, at angular slice
+ * `radial_offset` (Modes for concentric spherical shells are stored as
+ * consecutive blocks of angular modes). `l_max` must also be specified to
+ * determine the representation details.
+ */
+template <int Spin>
+std::complex<double> libsharp_mode_to_goldberg(
```

Comment:
to be consistent with the rest of the "spectral" code, probably `libsharp_mode_to_goldberg` and `goldberg_to_libsharp_mode` would be the preferred names (as opposed to a `from`). For example, the orthogonal polynomials use `modal_to_nodal` and `nodal_to_modal`